pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -400,6 +404,9 @@ func (m *vmMutator) getNodeSelectorRequirementFromNetwork(defaultNamespace strin
 
 	nad, err := m.nad.Get(namespace, name)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
For issue https://github.com/harvester/harvester/issues/8109, changes on L407~409 are un-related.

could we only add L89~L92 for this PR? thanks.

Reply:
Updated it. Thanks.